examples/regex-example.ts

```diff
@@ -40,85 +42,56 @@ const charmax10 = {
   message: 'Value must be 10 or fewer characters',
 }
 
-const FIELDS = {
-  supplierID: anmax50,
-  MBI: anmax11,
-  MemberZip: numexact5,
-  NPI: anmax15,
-  MemberAddress1: allcharmax50,
-  MemberAddress2: anmax50,
-  MemberAlias: anmax50,
-  MemberCity: lettersmax50,
-  Ending_DOS: charmax10,
-}
-
-const logger = { info: console.log }
-
-const fieldTypeVal = (record) => {
-  Object.keys(FIELDS).forEach((field) => {
-    const fieldRegex = FIELDS[field]['regex']
-    logger.info(FIELDS[field]['message'])
-    logger.info(fieldRegex.toString())
-    const fieldValue = record.get(field)
-    if (fieldRegex.test(fieldValue) === false) {
-      record.addError(field, FIELDS[field]['message'])
-      logger.info(FIELDS[field]['message'])
+const regexValidate = (regexMessage) => {
+  const retFunc = (val:string) => {
+    const regex = regexMessage.regex
+    const message = regexMessage.message
+    if (regex.test(val) === false) {
+      logger.info(regex.toString())
+      logger.info(message)
+      return [
+        new Message(message, 'error', 'validate')
+          ]
     }
-  })
+  }
+  return retFunc
 }
 
+
 export const SupplierMember = new Sheet(
   'SupplierMember',
   {
-
     supplierID: TextField({
       label: 'supplierID',
+      validate: regexValidate(anmax50)
     }),
 
     NPI: TextField({
-      label: 'NPI',
+      label: 'NPI', validate:regexValidate(anmax15)
     }),
 
     MBI: TextField({
-      label: 'MBI',
+      label: 'MBI', validate:regexValidate(anmax11)
     }),
 
-    MemberAddress1: TextField({
-      label: 'MemberAddress1',
-    }),
-
-    MemberAddress2: TextField({
-      label: 'MemberAddress2',
-    }),
+    MemberAddress1: TextField({label: 'MemberAddress1', validate: regexValidate(allcharmax50)}),
+    MemberAddress2: TextField({label: 'MemberAddress2', validate: regexValidate(anmax50)}),
```

Comment:
Look at how you can see exactly which validation is being applied to each field right here in one place.  This is much easier to undertand than the previous method.